lib/glific/flows/broadcast.ex

```diff
@@ -42,20 +42,39 @@ defmodule Glific.Flows.Broadcast do
 
     group_messages =
       Enum.map(group_ids, fn group_id ->
-        {:ok, group} = Repo.fetch_by(Group, %{id: group_id})
+        case Repo.fetch_by(Group, %{id: group_id}) do
+          {:ok, group} ->
+            create_group_message(flow, group)
 
-        {:ok, group_message} =
-          Messages.create_group_message(%{
-            body: "Starting flow: #{flow.name} for group: #{group.label}",
-            type: :text,
-            group_id: group_id
-          })
+          {:error, _reason} ->
+            {:error, "Group not found with ID: #{group_id}"}
+        end
+      end)
```

Comment:
I think if we use `Enum.reduce/3` instead of `Enum.map/2` then we can just accumulate the group_messages by appending it as a list in ln:47, and if group  doesn't exist then just return group_messages list itself. This way we can eliminate the need for an extra `Enum.filter/2` in ln:54th line